caffe2/opt/optimize_ideep.cc

```diff
@@ -0,0 +1,46 @@
+#include "caffe2/opt/optimize_ideep.h"
+#include "caffe2/opt/converter.h"
+#include "caffe2/opt/fuse_conv_relu.h"
+
+namespace caffe2 {
+namespace opt {
+
+using namespace nom;
+caffe2::NetDef OptimizeForIdeep(caffe2::NetDef net) {
+  auto nn = convertToNNModule(net);
+
+  // Conv+Relu fusion
+  auto should_fuse = [](const repr::Conv& conv) {
+    const auto annotation = conv.getAnnotation();
+    if (!annotation || !isa<Caffe2Annotation>(annotation)) {
+      return false;
+    }
+    const auto* op = dyn_cast<Caffe2Annotation>(annotation)->getOperatorDef();
+
+    // We only want to fuse for fast NNPACK convs
+    if (op->device_option().device_type() != DeviceType::IDEEP) {
+      return false;
+    }
+
+    return true;
+  };
+  auto postprocess = [](repr::NNGraph::NodeRef conv_node) {
+    auto conv = repr::nn::get<repr::Conv>(conv_node);
+    auto annotation = conv->getMutableAnnotation();
+    if (!annotation || !isa<Caffe2Annotation>(annotation)) {
+      return;
+    }
+    auto* op = dyn_cast<Caffe2Annotation>(annotation)->getMutableOperatorDef();
```

Comment:
we should probably abstract this step (in `caffe2/opt/convert.h` I guess) as it is quite commonly used
I'll get that up in a different diff